hass-addon-sunsynk-dev/mqtt.py

```diff
@@ -63,14 +63,12 @@ class Entity:
     expire_after: int = attr.field(default=0)  # unavailable if not updated
     enabled_by_default: bool = attr.field(default=True)
     entity_category: str = attr.field(default="")
+    icon: str = attr.field(default="")
 
     _path = ""
 
     def __attrs_post_init__(self) -> None:
         """Init the class."""
-        if not self.device_class:
-            # Try device_class from unit_of_measurement
-            self.device_class = hass_device_class(unit=self.unit_of_measurement)
         if not self.state_class and self.device_class == "energy":
```

Comment:
since we do not set the device_class, we might now miss total_increasing...

```suggestion
        if not self.state_class and self.unit_of_measurement == "kWh":
```

Reply:
I thought about this but we are still setting `device_class` for the regular (non-RW) sensors, so by the time `__attrs_post_init__` runs, there would be a `device_class` set?

Reply:
I double checked, the `device_class` and `state_class` fields still get set for the non-RW sensors:
<img width="244" alt="image" src="https://user-images.githubusercontent.com/16413523/186996349-478ddf4d-ad17-4def-83fb-d19371c94b8c.png">

Reply:
Thx for checking!